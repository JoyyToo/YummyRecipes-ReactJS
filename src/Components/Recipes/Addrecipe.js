import React, { Component }  from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import TextField  from 'material-ui/TextField';
import Paper from 'material-ui/Paper';
import { Link } from 'react-router-dom';
import {notify} from 'react-notify-toast'
import axiosInstance from '../Constants/AxiosCall';
import '../styles.css'


class AddRecipe extends Component {
    // initialize state
    state = {
        name: '',
        time: '',
        ingredients: '',
        procedure: '',
        id: '',
        category_id: '',
    }

    // handle user input
    handleInputChange = (event) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({[name]: value});
    }

    // mount token when page loads
    componentDidMount() {
        const token = window.localStorage.getItem('token');
        if (!token) {
            window.location.replace('/login')
        }
    }

    // handle add recipe request
    handleAddRecipe = (event) => {
        const category_id = this.props.match.params['category_id'];
        const payload = new FormData()
        payload.set('name', this.state.name)    
        payload.set('time', this.state.time,)
        payload.set('ingredients', this.state.ingredients,)
        payload.set('procedure', this.state.procedure,)
        
        const token = window.localStorage.getItem('token')
        if (!token) {
            window.location.replace('/login')
        }
            // send POST request to API
            axiosInstance.post(`category/${category_id}/recipes`, payload)
            .then((response) => {
                notify.show(response.data.message, 'success', 4000);
                this.props
                .history
                .push(`/categories/${category_id}/recipes`);
            })
            .catch((error) => {
                notify.show(error.response.data.message, 'error', 4000);
            });
        }

    // render add recipe form
    render(){
        const id = this.state.id
        const category_id = this.props.match.params['category_id'];

        let style = {
            marginLeft: 20,
            width: 440,
          };
        let divstyle = {
            width: 500,
            margin: 'auto',
            textAlign: 'left',
            paddingLeft: 20,
        }
        return (
            <MuiThemeProvider>
            <div className="main">
                <div className="inputs" style={{paddingTop: '60px'}}>
                <Paper zDepth={2} style={divstyle}>
                    <p class="heading"><b>ADD RECIPE</b></p>

                    <TextField floatingLabelText="Name" name="name" 
                                 value={this.state.name} style={style} onChange={this.handleInputChange}/><br /><br />
                                 
                    <TextField floatingLabelText="Time taken" name="time" 
                                 value={this.state.time} style={style} onChange={this.handleInputChange}/><br /><br />

                    <TextField floatingLabelText="Ingredients" name="ingredients" 
                                 value={this.state.ingredients} style={style} multiLine={true} rows={3}
                                 onChange={this.handleInputChange} /><br />

                    <TextField floatingLabelText="Procedure" name="procedure" 
                                 value={this.state.procedure} style={style} multiLine={true} rows={3}
                                 onChange={this.handleInputChange} /><br />

                    <div className="buttons" id="btnstyle">
                    <button className="network" id="one" onClick={(event => this.handleAddRecipe(event))}>ADD</button>
                    <Link to={"/categories/" + category_id + "/recipe/" + id}>
                    <button className="network" id="two">CANCEL</button></Link>

                    </div>

                    <p>A recipe has no soul. You are the cook, give life to the recipe</p> <br />

                </Paper><br />
                    
                </div>
            </div>
            </MuiThemeProvider>
        );
    }}

export default AddRecipe;