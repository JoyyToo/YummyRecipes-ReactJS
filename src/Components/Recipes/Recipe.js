import React, { Component } from 'react';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Card, CardText } from 'material-ui/Card';
import AutoComplete from 'material-ui/AutoComplete';
import FlatButton from 'material-ui/FlatButton';
import { Link } from 'react-router-dom';
import { notify } from 'react-notify-toast';
import axiosInstance from '../Constants/AxiosCall';
import Pagination from '../Pagination/PaginationApp';

class Recipe extends Component {
  // initialize state
  constructor(props) {
    super(props);
    this.state = {
      pagination: '',
      recipes: [],
      category_id: '',
      id: '',
      q: '',

    };
  }

  // call handle recipe before mounting
  componentDidMount() {
    this.handleRecipe();
  }

  // handle user input
  handleInputChange = (event) => {
    this.setState({ q: event });
  };

  // handle get recipe request
  handleRecipe = (page = 1) => {
    const categoryid = this.props.match.params['category_id'];

    // send GET request to API
    axiosInstance.get(`category/${categoryid}/recipes?page=${page}`)
      .then((response) => {
        this.setState({
          recipes: response.data[1],
          id: this.state.id,
          category_id: this.state.category_id,
          pagination: response.data[0],
        });
      })

      .catch((error) => {
        if (error.response) {
          notify.show(error.response.data.message, 'error', 4000);
        }
      });
  };

  // handle delete recipe request
  handleDeleteRecipe = (event) => {
    const id = event.currentTarget.getAttribute('id');
    const categoryid = this.props.match.params['category_id'];

    // send DELETE request to API
    axiosInstance.delete(`category/${categoryid}/recipes/${id}`)
      .then((response) => {
        this.handleRecipe();
        notify.show(response.data.message, 'success', 4000);
        this.setState({
          id: this.state.id,
        });
        this.props
          .history
          .push(`/categories/${categoryid}/recipes`);
      })
      .catch((error) => {
        notify.show(error.response, 'error', 4000);
      });
  };

  // handle search recipe request
  handleSearch = () => {
    const categoryid = this.props.match.params['category_id'];

    // send GET request to API
    axiosInstance.get(`category/${categoryid}/recipes?q=${this.state.q}`)
      .then((response) => {
        const recipe = response.data['recipes'];
        this.setState({
          q: '',
          recipes: recipe,
        });
      })
      .catch((error) => {
        if (error.response) {
          notify.show(error.response.data.message, 'error', 4000);
        }
      });
  };

  // validate pagination
  button() {
    const recipe = this.state.recipes;

    if (recipe !== '') {
      if (this.state.pagination['current page'] === 1) {
        if (recipe.length > 5) {
          return (
            <footer>
              <Pagination className="page" changePage={this.handleRecipe.bind(this)} paginationObject={this.state.pagination} />
            </footer>
          );
        }
      } else if (this.state.pagination['current page'] !== 1) {
        if (recipe.length >= 1) {
          return (
            <footer>
              <Pagination className="page" changePage={this.handleRecipe.bind(this)} paginationObject={this.state.pagination} />
            </footer>
          );
        }
      }
    }
  }

  // render recipes
  render() {
    const recipe = this.state.recipes;

    const style = {
      marginLeft: 20,
    };

    // handle get recipe request
    handleRecipe = (page=1) => {
        const category_id = this.props.match.params['category_id'];
    }
    // mount token when page loads
    componentDidMount() {
        this.setState({
            token: window.sessionStorage.accessToken,
        });
        const token = window.localStorage.getItem('token');
        if (!token) {
            window.location.replace('/login')
        }

        // send GET request to API
        axiosInstance.get(`category/${category_id}/recipes?page=${page}`)
            .then((response) => {
                this.setState({
                    data: response.data,
                    recipes: response.data[1],
                    id: this.state.id,
                    category_id: this.state.category_id,
                    pagination: response.data[0]
                })
            })

            .catch((error) => {
                if (error.response){
                    notify.show(error.response.data.message,'error', 4000);
                }
            });
    };

      },
      align: {
        align: 'left',
      

    // handle delete recipe request
    handleDeleteRecipe = (event) => {
        let id = event.currentTarget.getAttribute('id');
        const category_id = this.props.match.params['category_id'];

        // send DELETE request to API
        axiosInstance.delete(`category/${category_id}/recipes/${id}`)
            .then((response) => {
                this.handlerecipe();

        const token = window.localStorage.getItem('token');
        // send DELETE request to API
        axiosInstance.delete(`category/${category_id}/recipes/${id}`)
            .then((response) => {
                this.handlerecipe();
                notify.show(response.data.message, 'success', 4000);
                this.setState({
                    id: this.state.id
                });
                this.props
                    .history
                    .push(`/categories/${category_id}/recipes`);
            })
            .catch((error) => {
                notify.show(error.response, 'error', 4000);
            });
    });


    handleSearch= (event) => {
        const category_id = this.props.match.params['category_id'];

        // send GET request to API
        axiosInstance.get(`category/${category_id}/recipes?q=${this.state.q}`)
            .then((response) => {
                let recipe = response.data['recipes'];
                this.setState({
                        q: "",
                        recipes: recipe,
                });   
            })
            .catch((error) => {
                notify.show(error.response.data.message, 'error', 4000);
            });
    };

        // send GET request to API
        axiosInstance.get(`category/${category_id}/recipes?q=${this.state.q}`)
            .then((response) => {
                let recipe = response.data['recipes'];
                this.setState({
                        q: "",
                        recipes: recipe,
                });   
            })
            .catch((error) => {
                notify.show(error.response.data.message, 'error', 4000);
            });
    };

    render() {
        let recipe = this.state.recipes;

        let style = {
            marginLeft: 20,
        };

        const styles = {
            card2: {
                position: 'relative',
                padding: 10,
                width: 450,
                float: 'left',
                margin: 5,

            },
            align: {
                align: 'left'

            }
        };

        const category_id = this.props.match.params['category_id'];
        return (
            <div>
                
                <MuiThemeProvider>
                    <div>
                        <Link to={"/categories/"+ category_id +"/addrecipe"}>
                            <FloatingActionButton secondary={true} style={style}>
                                <ContentAdd/>
                            </FloatingActionButton>
                        </Link>

                        <SearchBar name="q" value={this.state.q}
                            onChange={this.handleInputChange}
                            onRequestSearch={this.handleSearch.bind(this)}

                            style={{
                            margin: '0 auto',
                            width: 430,
                            float: 'right',
                            marginRight: 75,
                            backgroundColor: 'white',
                            paddingLeft: 20                  
                            }}
                        />
                      </Link>
                      <FlatButton
                        label="Delete"
                        style={{ color: 'red' }}
                        category_id={recipe[key]['category_id']}
                        id={recipe[key]['id']}
                        onClick={(event => this.handleDeleteRecipe(event))}
                      />
                    </Card>

                  </MuiThemeProvider>

                </div>

              ))}

          </div>

        </MuiThemeProvider>
        {this.button()}
      </div>

    );
  }
}

export default Recipe;
