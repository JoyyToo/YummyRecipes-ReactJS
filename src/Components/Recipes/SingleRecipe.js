import React, {Component} from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {Card, CardText,} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import {Link} from 'react-router-dom';
import {notify} from 'react-notify-toast'
import axiosInstance from '../Constants/AxiosCall';

class SingleRecipe extends Component {
    // initialize state
    constructor(props) {
        super(props);
        this.state = {
            recipes: [],
            id: "",
            date_created: '',
            date_modified: '',
            name: '',
            time: '',
            ingredients: '',
            procedure: '',
            category_id: '',

        }
    }

    // mount recipes when page loads
    componentDidMount() {
        this.handleRecipe();
    }

    // handle get recipe request
    handleRecipe = (props) => {
        const id = this.props.match.params['id'];
        const category_id = this.props.match.params['category_id'];

        // send GET request to API
        axiosInstance.get(`category/${category_id}/recipes/${id}`)
            .then((response) => {
                let recipe = response.data['recipes'];
                this.setState({
                    date_created: recipe['date_created'],
                    date_modified: recipe['date_modified'],
                    time: recipe['time'],
                    ingredients: recipe['ingredients'],
                    procedure: recipe['procedure'],                                      
                    name: recipe['name'],
                    category_id: recipe['category_id'],
                    id: recipe['id'],
                })
            })

            .catch((error) => {
                notify.show(error.response, 'error', 4000);
            });
    };


    // handle delete recipe request
    handleDeleteRecipe = (event) => {
        let id = event.currentTarget.getAttribute('id');
        const category_id = this.props.match.params['category_id'];

        // send DELETE request to API
        axiosInstance.delete(`category/${category_id}/recipes/${id}`)
            .then((response) => {
                this.handleRecipe();
                notify.show(response.data.message, 'success', 4000);
                this.setState({
                    id: this.state.id
                });
                this.props
                    .history
                    .push(`/categories/${category_id}/recipes`);
            })

            .catch((error) => {
                notify.show(error.response, 'error', 4000);
            });
    };

    // render a single recipe
    render() {
        let recipe = this.state;

        const styles = {
            card2: {
                position: 'relative',
                padding: 10,
                width: 450,
                float: 'left',
                margin: 5,

            },
            align: {
                align: 'left'

            }
        };
        return (
            <div>
                <MuiThemeProvider>

                    <div>
                        {(
                            <div style={styles.card2}>
                                <MuiThemeProvider is="nospace start">
                                    <Card >
                                        <h2 style={{marginLeft: 30, paddingTop: 10}}>{recipe['name']}</h2>
                                        <p style={{marginLeft: 30, marginBottom: 30}}>Time taken: {recipe['time']}</p>
                                        <p style={{marginLeft: 30, marginBottom: 30}}>Ingridients: {recipe['ingredients']}</p>
                                        <p style={{marginLeft: 30, marginBottom: 30}}>Procedure: {recipe['procedure']}</p>
                                        <CardText>
                                            {
                                                <div>

                                                    <p style={{marginLeft: 15}}><i>Created:</i> {this.state.date_created}</p>
                                                    <p style={{marginLeft: 15, marginBottom: 30}}><i>Modified:</i> {this.state.date_modified}</p>

                                                </div>

                                            }
                                        </CardText>
                                     <Link to={"/categories/" + recipe['category_id'] + "/recipes/" + recipe['id']}>
                                                <FlatButton label="Edit"
                                                            id={recipe['id']} desc={recipe['desc']}
                                                            primary={true}/></Link>
                                            <FlatButton label="Delete" style={{'color': 'red'}}
                                                        category_id={recipe['category_id']}
                                                        id={recipe['id']}
                                                        onClick={(event => this.handleDeleteRecipe(event))} />
                                    </Card>

                                </MuiThemeProvider>
                            </div>
                        )}
                    </div>

                </MuiThemeProvider>
            </div>

        )
    }

};

export default SingleRecipe;

