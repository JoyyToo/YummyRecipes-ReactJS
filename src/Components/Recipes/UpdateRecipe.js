import React, { Component }  from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import TextField  from 'material-ui/TextField';
import Paper from 'material-ui/Paper';
import { Link } from 'react-router-dom';
import {notify} from 'react-notify-toast'
import axiosInstance from '../Constants/AxiosCall';
import '../styles.css'


class UpdateRecipe extends Component {
    // initialize state
    state = {
        name: '',
        time: '',
        ingredients: '',
        procedure: '',
        id: '',
        category_id: '',
    }

    // handle user input
    handleInputChange = (event) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({[name]: value});
    };

    // handle get recipe request
    getRecipe() {
        const id = this.props.match.params['id'];
        const category_id = this.props.match.params['category_id'];
        const token = window.localStorage.getItem('token');
        if (!token) {
            window.location.replace('/login')
        }

        // send GET request to API
        axiosInstance.get(`category/${category_id}/recipes/${id}`)
            .then((response) => {
                let recipe = response.data['recipes'];
                this.setState({
                    id: id,
                    category_id: category_id,
                    name: recipe['name'],
                    time: recipe['time'],
                    ingredients: recipe['ingredients'],
                    procedure: recipe['procedure'],
                })
            })
            .catch((error) => {
                notify.show(error.response, 'error', 4000);
            });
    }

    // handle edit recipe request
    handleEditRecipe = () => {
        const payload = new FormData();
        payload.set('name', this.state.name)    
        payload.set('time', this.state.time,)
        payload.set('ingredients', this.state.ingredients,)
        payload.set('procedure', this.state.procedure,)

        // send PUT request to API
        axiosInstance.put(`category/${this.state.category_id}/recipes/${this.state.id}`, payload)
            .then((response) => {
                notify.show(response.data.message, 'success', 4000);
                this.props
                    .history
                    .push(`/categories/${this.state.category_id}/recipes`);
            })

            .catch((error) => {
                notify.show(error.response.data.message, 'error', 4000);
              
            });
    };

    // mount recipes and when page loads
    componentDidMount() {
        this.getRecipe();
    }
    
    // render edit recipe form
    render(){
        let style = {
            marginLeft: 20,
            width: 440,
          };
        let divstyle = {
            width: 500,
            margin: 'auto',
            textAlign: 'left',
            paddingLeft: 20,
        }
        return (
            <MuiThemeProvider>
            <div className="main">
                <div className="inputs" style={{paddingTop: '60px'}}>
                <Paper zDepth={2} style={divstyle}>
                    <p class="heading"><b>UPDATE RECIPE</b></p>

                    <TextField floatingLabelText="Name" name="name" 
                                 value={this.state.name} style={style} onChange={this.handleInputChange} /><br /><br />

                    <TextField floatingLabelText="Time taken" name="time" 
                                 value={this.state.time} style={style} onChange={this.handleInputChange} /><br /><br />

                    <TextField floatingLabelText="Ingridients" name="ingredients" 
                                 value={this.state.ingredients} style={style} multiLine={true} rows={3}
                                 onChange={this.handleInputChange} /><br />

                    <TextField floatingLabelText="Procedure" name="procedure" 
                                 value={this.state.procedure} style={style} multiLine={true} rows={3} 
                                 onChange={this.handleInputChange} /><br />

                    <div className="buttons" id="btnstyle">
                    <button className="network" id="one" onClick={(event => this.handleEditRecipe(event))}>UPDATE</button>
                    <Link to={"/categories/" + this.state.category_id + "/recipe/" + this.state.id}>
                    <button className="network" id="two">CANCEL</button></Link>

                    </div>

                    <p>A recipe has no soul. You are the cook, give life to the recipe</p> <br />

                </Paper><br />
                    
                </div>
            </div>
            </MuiThemeProvider>
        );
    }}

export default UpdateRecipe;
