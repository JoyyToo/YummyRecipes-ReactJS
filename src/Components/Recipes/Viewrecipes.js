import React, {Component} from 'react';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
<<<<<<< HEAD
import {Card, CardText} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import {Link} from 'react-router-dom';
import {notify} from 'react-notify-toast'
import axiosInstance from '../Constants/Axioscall';
import SearchBar from 'material-ui-search-bar'
import Pagination from '../Pagination/Pagination'

class Recipe extends Component {
    // initialize state
    constructor(props) {
        super(props);
        this.state = {
            pagination: "",
=======
import {Card, CardText,} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import {Link} from 'react-router-dom';
import {notify} from 'react-notify-toast'
import axios from 'axios';
import * as constant from "../constant";
import SearchBar from 'material-ui-search-bar'
import Pagination from 'material-ui-pagination';

class Recipe extends Component {
    constructor(props) {
        super(props);
        this.state = {
>>>>>>> [feature] Add recipe components
            recipes: [],
            data: "",
            category_id: "",
            id: "",
<<<<<<< HEAD
            q: "",
=======
>>>>>>> [feature] Add recipe components

        }
    }

<<<<<<< HEAD
    // handle user input
    handleInputChange = (event) => {
        this.setState({q: event});    

    };

    // handle get recipe request
    handlerecipe = (page=1) => {
        const category_id = this.props.match.params['category_id'];
=======

    componentDidMount() {
        this.setState({
            token: window.sessionStorage.accessToken,
        });
>>>>>>> [feature] Add recipe components
        const token = window.localStorage.getItem('token');
        if (!token) {
            window.location.replace('/login')
        }
<<<<<<< HEAD

        // send GET request to API
        axiosInstance.get(`category/${category_id}/recipes?page=${page}`)
            .then((response) => {
                let recipe = response.data[1];
                let paginationObject = response.data[0];
=======
    }

    handlerecipe = (props) => {
        const category_id = this.props.match.params['category_id'];
        const token = window.localStorage.getItem('token');
        axios({
            url: `${constant.URL}/category/${category_id}/recipes`,
            method: 'get',
            headers: {
                Authorization: `Bearer ${token}`,
                'Content-Type': 'multipart/form-data',
            }

        })

            .then((response) => {
                let recipe = response.data[1];
>>>>>>> [feature] Add recipe components

                this.setState({
                    data: response.data,
                    recipes: recipe,
                    id: this.state.id,
                    category_id: this.state.category_id,
<<<<<<< HEAD
                    pagination: paginationObject
=======
>>>>>>> [feature] Add recipe components
                })
            })

            .catch((error) => {
                notify.show(error.response.data.message, 'error', 4000);
            });
    };

<<<<<<< HEAD
    // call handle recipe before mounting
=======
>>>>>>> [feature] Add recipe components
    componentWillMount() {
        this.handlerecipe();
    }

<<<<<<< HEAD
    // handle delete recipe request
=======
>>>>>>> [feature] Add recipe components
    handleDeleterecipe = (event) => {
        let id = event.currentTarget.getAttribute('id');
        const category_id = this.props.match.params['category_id'];

<<<<<<< HEAD
        // send DELETE request to API
        axiosInstance.delete(`category/${category_id}/recipes/${id}`)
            .then((response) => {
                this.handlerecipe();
=======
        let th = this;

        const token = window.localStorage.getItem('token');

        axios({
            url: `${constant.URL}/category/${category_id}/recipes/${id}`,
            method: 'delete',
            headers: {
                Authorization: `Bearer ${token}`,
                'Content-Type': 'multipart/form-data',
            }
        })

            .then((response) => {

                th.handlerecipe();

>>>>>>> [feature] Add recipe components
                notify.show(response.data.message, 'success', 4000);
                this.setState({
                    id: this.state.id
                });
                this.props
                    .history
                    .push(`/categories/${category_id}/recipes`);
            })
<<<<<<< HEAD
=======

>>>>>>> [feature] Add recipe components
            .catch((error) => {
                notify.show(error.response, 'error', 4000);
            });
    };

<<<<<<< HEAD
    // handle search recipe request
    handleSearch= (event) => {
        const category_id = this.props.match.params['category_id'];

        // send GET request to API
        axiosInstance.get(`category/${category_id}/recipes?q=${this.state.q}`)
            .then((response) => {
                let recipe = response.data['recipes'];
                this.setState({
                        q: "",
                        recipes: recipe,
                });   
            })
            .catch((error) => {
                notify.show(error.response.data.message, 'error', 4000);
            });
    };

    // render recipes
=======

>>>>>>> [feature] Add recipe components
    render() {
        let recipe = this.state.recipes;

        let style = {
            marginLeft: 20,
        };

        const styles = {
            card2: {
                position: 'relative',
                padding: 10,
                width: 450,
                float: 'left',
                margin: 5,

            },
            align: {
                align: 'left'

            }
        };

        const category_id = this.props.match.params['category_id'];
<<<<<<< HEAD
=======

>>>>>>> [feature] Add recipe components
        return (
            <div>
                
                <MuiThemeProvider>
                    <div>
                        <Link to={"/categories/"+ category_id +"/addrecipe"}>
                            <FloatingActionButton secondary={true} style={style}>
                                <ContentAdd/>
                            </FloatingActionButton>
                        </Link>
<<<<<<< HEAD
                        <SearchBar name="q" value={this.state.q}
                            onChange={this.handleInputChange}
                            onRequestSearch={this.handleSearch.bind(this)}

                            style={{
                                margin: '0 auto',
                                width: 430,
                                float: 'right',
                                marginRight: 75,
                            }}
=======
                        <SearchBar
                          onChange={() => console.log('onChange')}
                          onRequestSearch={() => console.log('onRequestSearch')}
                          style={{
                            margin: '0 auto',
                            width: 430,
                              float: 'right',
                              marginRight: 75,
                          }}
>>>>>>> [feature] Add recipe components
                        />
                    </div>

                    <div>
                        
                        {

                            Object.keys(recipe).map((key) => (
                                <div style={styles.card2} key={key}>
                                    <MuiThemeProvider is="nospace start">
                                        <Card style={{height: 250}}>
                                            <h2 style={{marginLeft: 30, 
                                                paddingTop: 10}} >
                                                {recipe[key]['name']}</h2>
                                            <p style={{marginLeft: 30}}>Time: {recipe[key]['time']}</p>
                                            <CardText>
                                                {

                                                }
                                            </CardText>
<<<<<<< HEAD

                                            <Link to={"/categories/" + recipe[key]['category_id'] + "/recipe/" + recipe[key]['id']}>
                                            <FlatButton label="View"
                                                        category_id={recipe[key]['category_id']}
                                                        id={recipe[key]['id']}
                                            /></Link>               
=======
                            
>>>>>>> [feature] Add recipe components
                                            <Link to={"/categories/" + recipe[key]['category_id'] + "/recipes/" + recipe[key]['id']}>
                                                <FlatButton label="Edit"
                                                            id={recipe[key]['id']} desc={recipe[key]['desc']}
                                                            primary={true}/></Link>
                                            <FlatButton label="Delete" style={{'color': 'red'}}
                                                        category_id={recipe[key]['category_id']}
                                                        id={recipe[key]['id']}
                                                        onClick={(event => this.handleDeleterecipe(event))}
                                            />
<<<<<<< HEAD
=======
                                            <Link to={"/categories/" + recipe[key]['category_id'] + "/recipe/" + recipe[key]['id']}>
                                            <FlatButton label="View" primary={true}
                                                        category_id={recipe[key]['category_id']}
                                                        id={recipe[key]['id']}
                                            /></Link>
>>>>>>> [feature] Add recipe components
                                        </Card>

                                    </MuiThemeProvider>

                                </div>
                                            
                            ))}
<<<<<<< HEAD
                            <Pagination changePage={this.handlerecipe.bind(this)} paginationObject={this.state.pagination}/>
=======
>>>>>>> [feature] Add recipe components
                          
                    </div>

                </MuiThemeProvider>

            </div>

        )
    }

};

export default Recipe;

