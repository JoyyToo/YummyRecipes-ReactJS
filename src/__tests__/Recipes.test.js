import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Recipes from '../Components/Recipes/Viewrecipes'

describe('Recipes component', () => {
    const params = {
        match: {
            params: {
                id:1
            }
        }
    }
    const wrapper = shallow(<Recipes match={{params}}/>);
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('it renders handlerecipe correctly', () =>{
        expect(wrapper.instance().handlerecipe( {preventDefault} ))    
    });

    it('it renders handleInputChange correctly', () =>{
        expect(wrapper.instance().handleInputChange( {preventDefault} ))    
    });

})