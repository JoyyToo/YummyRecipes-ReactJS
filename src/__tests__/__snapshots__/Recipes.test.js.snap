// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes component renders properly 1`] = `
<div>
  <MuiThemeProvider>
    <div>
      <Link
        replace={false}
        to="/categories/undefined/addrecipe"
      >
        <FloatingActionButton
          disabled={false}
          mini={false}
          secondary={true}
          style={
            Object {
              "marginLeft": 20,
            }
          }
          zDepth={2}
        >
          <ContentAdd />
        </FloatingActionButton>
      </Link>
      <SearchBar
        closeIcon={
          <NavigationClose
            color="#9e9e9e"
          />
        }
        dataSource={Array []}
        dataSourceConfig={
          Object {
            "text": "text",
            "value": "value",
          }
        }
        disabled={false}
        hintText="Search"
        name="q"
        onChange={[Function]}
        onRequestSearch={[Function]}
        searchIcon={
          <ActionSearch
            color="#9e9e9e"
          />
        }
        spellCheck={false}
        style={
          Object {
            "float": "right",
            "margin": "0 auto",
            "marginRight": 75,
            "width": 430,
          }
        }
        value=""
      />
    </div>
    <div />
  </MuiThemeProvider>
  <footer>
    <PaginationApp
      changePage={[Function]}
      paginationObject=""
    />
  </footer>
</div>
`;
